sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -406,6 +406,160 @@ def _upload_blob_options(  # pylint:disable=too-many-statements
+            kwargs['blob_headers'] = BlobHTTPHeaders(
+                blob_cache_control=content_settings.cache_control,
+                blob_content_type=content_settings.content_type,
+                blob_content_md5=bytearray(content_settings.content_md5) if content_settings.content_md5 else None,
+                blob_content_encoding=content_settings.content_encoding,
+                blob_content_language=content_settings.content_language,
+                blob_content_disposition=content_settings.content_disposition
+            )
+        cpk = kwargs.pop('cpk', None)
+        cpk_info = None
+        if cpk:
+            if self.scheme.lower() != 'https':
+                raise ValueError("Customer provided encryption key must be used over HTTPS.")
+            cpk_info = CpkInfo(encryption_key=cpk.key_value, encryption_key_sha256=cpk.key_hash,
+                               encryption_algorithm=cpk.algorithm)
+        kwargs['cpk_info'] = cpk_info
+        kwargs['cpk_scope_info'] = get_cpk_scope_info(kwargs)
+
+        options = {
+            'copy_source_blob_properties': copy_source_blob_properties,
+            'copy_source': source_url,
+            'timeout': timeout,
+            'modified_access_conditions': dest_mod_conditions,
+            'blob_tags_string': blob_tags_string,
+            'headers': headers,
+            'cls': return_response_headers,
+        }
+        source_mod_conditions = get_source_conditions(kwargs)
+        dest_access_conditions = get_access_conditions(kwargs.pop('destination_lease', None))
+        options['source_modified_access_conditions'] = source_mod_conditions
+        options['lease_access_conditions'] = dest_access_conditions
+        options['tier'] = tier.value if tier else None
+        options.update(kwargs)
+        return options
+
+    @distributed_trace
+    def upload_blob_from_url(
+            self, source_url,   # type: str
+            copy_source_blob_properties=True,   # type: Optional[bool]
+            **kwargs):
+        # type: (...) -> Dict[str, Any]
+        """
+        Creates a new Block Blob where the content of the blob is read from a given URL.
+        The content of an existing blob is overwritten with the new blob.
+
+        :param str source_url:
+            A URL of up to 2 KB in length that specifies a file or blob.
+            The value should be URL-encoded as it would appear in a request URI.
+            If the source is in another account, the source must either be public
+            or must be authenticated via a shared access signature. If the source
+            is public, no authentication is required.
+            Examples:
+            https://myaccount.blob.core.windows.net/mycontainer/myblob
+
+            https://myaccount.blob.core.windows.net/mycontainer/myblob?snapshot=<DateTime>
+
+            https://otheraccount.blob.core.windows.net/mycontainer/myblob?sastoken
+        :param bool copy_source_blob_properties:
```

Comment:
Does other platforms also call this copy_*? Feel it’s like for a copy operation

Reply:
This is not available in other apis. This is only specific to the new API. This is one of the parameters the generated code accepts https://github.com/Azure/azure-sdk-for-python/blob/46362a0d75d088c647fa2131d2ef3062db9214b7/sdk/storage/azure-storage-blob/azure/storage/blob/_generated/aio/operations_async/_block_blob_operations_async.py#L242